src/eradiate/scenes/spectra/_air_scattering_coefficient.py

```diff
@@ -46,63 +34,37 @@ def eval_mono(self, w: pint.Quantity) -> pint.Quantity:
         # Inherit docstring
         return compute_sigma_s_air(wavelength=w)
 
-    def eval_ckd(self, *bindexes: Bindex) -> pint.Quantity:
+    def eval_ckd(self, w: pint.Quantity, g: float) -> pint.Quantity:
         # Inherit docstring
-        # Spectrum is averaged over spectral bin
-
-        result = np.zeros((len(bindexes),))
-        quantity_units = ucc.get(self.quantity)
-
-        for i_bindex, bindex in enumerate(bindexes):
-            bin = bindex.bin
-
-            # -- Build a spectral mesh with spacing finer than 1 nm
-            #    (reasonably accurate)
-            wmin_m = bin.wmin.m_as(ureg.nm)
-            wmax_m = bin.wmax.m_as(ureg.nm)
-            w = np.linspace(wmin_m, wmax_m, 2)
-            n = 10
-
-            while True:
-                if w[1] - w[0] <= 1.0:  # nm
-                    break
-                w = np.linspace(wmin_m, wmax_m, n + 1)
-                n *= 2
-
-            w = w * ureg.nm
-
-            # -- Evaluate spectrum at wavelengths
-            interp = self.eval_mono(w)
-
-            # -- Average spectrum on bin extent
-            integral = np.trapz(interp, w)
-            result[i_bindex] = (integral / bin.width).m_as(quantity_units)
-
-        return result * quantity_units
+        return self.eval_mono(w)
 
     def integral(self, wmin: pint.Quantity, wmax: pint.Quantity) -> pint.Quantity:
         raise NotImplementedError
 
     @property
     def template(self) -> dict:
         # Inherit docstring
+
+        def eval(ctx: KernelDictContext) -> float:
+            kernel_units = uck.get("collision_coefficient")
+            return float(self.eval(ctx.si).m_as(kernel_units))
+
         return {
             "type": "uniform",
-            "value": InitParameter(
-                lambda ctx: float(
-                    self.eval(ctx.spectral_ctx).m_as(uck.get("collision_coefficient")),
-                )
-            ),
+            "value": InitParameter(evaluator=eval),
```

Comment:
I thought you had rolled this back, what happened here?

Reply:
I thought I had this rolled back too, perhaps I forgot to commit those changes